src/libexpr/parser.y

```diff
@@ -273,9 +273,14 @@ void yyerror(YYLTYPE * loc, yyscan_t scanner, ParseData * data, const char * err
   nix::Formal * formal;
   nix::NixInt n;
   nix::NixFloat nf;
-  const char * id; // !!! -> Symbol
-  char * path;
-  char * uri;
+  struct StringToken {
```

Comment:
Might be worth explaining why we can’t use a `string_view` here

Reply:
because the version of bison we use can't deal with it. we're planning on updating bison the 3.8 to get access to a proper C++ glr parser that *can* deal with that, but we're not entirely sure yet how to do that. currently it requires pulling *only* bison from a newer nixpkgs version than the flake uses because updating to a newer nixpkgs still breaks static build :slightly_frowning_face: